### DISCUSSION TITLE
Plugins do not work with newest version of Zim

### DISCUSSION THREAD

**Discussion by Nick222** [zim-desktop-wiki/zim-desktop-wiki#1609] [Q&A] [answered]
Good plugins - but do not work:

- seqdiag - Xubuntu 21.04 (but Raspberry Debian 10 - OK); (SEE SOLUTION)
- zeitgeist - Xubuntu 21.04 and Raspberry Debian 10; (NEED REMAKE)

Who can configure it ?

**Comment by introt** [comment]
Hi! Could you please elaborate on the way in which they do not work?

0. Are the dependencies, `libzeitgeist` (needs the `python-zeitgeist` package on Debian and Ubuntu) and `seqdiag`, installed?
    * According to the manual, `seqdiag` should be installed in the system path. What does the command `which seqdiag` return, on both systems? Is the returned path included in `$PATH`?
1. Do the plugins show up in the Preferences > Plugins? Which of the below screenshots do they look like, if any?

  *  A: ![image_A](https://user-images.githubusercontent.com/26069787/134331771-7881b70b-ba5a-41ff-90c7-11f424f2b2b9.png)
  _A: Usually indicative of a failure to parse the plugin file._

  * B: ![image_B](https://user-images.githubusercontent.com/26069787/134331886-cb7b654e-854f-4e48-9bd0-69048a3f10ea.png)
  _B: Failure due to dependencies failing._

  * C: ![image_C](https://user-images.githubusercontent.com/26069787/134331971-14bd47b1-2484-4606-b440-eeac96a07d64.png)
  _C: Failure due to all optional dependencies failing._

2. If the plugins can be enabled, but do not work in some other way, please describe what you're trying to do, providing screenshots, if possible.
    * Which applications are you using Zeitgeist with?
      * Do your other Zeitgeist-logging applications work as expected?
      * What's your desktop environment on Raspberry Debian?
3. Finally, please run Zim from the command line with the `--debug` argument, try to enable the plugins (open Preferences > Plugins) and paste the output.
    * (Alternatively, you can close Zim, delete `/tmp/zim-*`, open Zim the regular way and then post the contents from `/tmp/zim-*/zim.log`)

From your other thread I'm assuming you're running a version downloaded from the GitHub repository. Is this correct?

Thanks for your cooperation!

**Comment by Nick222** [comment]
1) About **seqdiag**:
![seqdiag](https://user-images.githubusercontent.com/307416/134359408-b12dc2a4-d375-4fc9-b0f7-5b6005e55971.png)
Where and how can I get seqdiag for Xubuntu 21.04 ?

2) After install new version gnome-activity-journal 1.0.0-3build1 - **zeitgeist**-plugin become work.
But Zim and gnome-activity-journal crashed...
What this plugin must do? I cannot see anything...

**Reply by introt** [reply]
1. According to [the manual](https://zim-wiki.org/manual/Plugins/Sequence_Diagram_Editor.html), `sudo pip install seqdiag` or `sudo easy_install seqdiag` should do the trick
2. Zeitgeist
    * The plugin makes your Zim page modifications show up in, for example, the [GNOME Activity Journal](https://launchpad.net/gnome-activity-journal) (there's nothing new to see on Zim's side).
      * For more information, please refer to [the plugin's manual page](https://zim-wiki.org/manual/Plugins/Log_events_with_Zeitgeist.html).
    * You might need to enable starting GNOME services in XFCE settings for Zeitgeist to work
      * Also make sure the Python bindings are installed: `sudo apt install python-zeitgeist` (or `python3-zeitgeist`, depending on your distribution)

It's hard to tell what's wrong without any logs. Could you please run both Zim and GNOME Activity Journal from the terminal and post the output?

**Reply by introt** [reply]
Or if you have the crash reporter installed, you can get the relevant information that way

**Reply by introt** [reply]
I had a look into GNOME Activity Journal and ~[the project doesn't seem too healthy](https://answers.launchpad.net/gnome-activity-journal/+question/668686)~ Edit: never mind, it has just [moved to Gitlab](https://gitlab.gnome.org/crvi/gnome-activity-journal).

Zeitgeist itself should still be alive and well, so if you want to use it to log your activity, I suggest taking a look at its [Arch Linux wiki article](https://wiki.archlinux.org/title/Zeitgeist).

**Comment by Nick222** [comment]
> Requirement already satisfied: seqdiag in /usr/lib/python3/dist-packages (2.0.0)

About crash - see attachment
[zim_crash.zip](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/7212528/zim_crash.zip)

**Reply by introt** [reply]
Weird. Thanks for the logs, I'll spin up a VM to investigate.

**Reply by introt** [reply]
I wasn't able to reproduce a crash, but I did get
`TypeError: on_page_changed() takes 2 positional arguments but 3 were given`, which originates from [here](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/zim/plugins/zeitgeist-logger.py#L87). Since the plugin hasn't changed since 2018, it seems likely to have broken by a change in Zim. I added a dummy argument to counter this, but since the plugin [doesn't](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/zim/plugins/zeitgeist-logger.py#L55) produce [nearly](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/zim/plugins/zeitgeist-logger.py#L59) enough logs for further debugging, I had to leave it there. Other applications' events showed up in GNOME Activity Journal just fine.

**Comment by introt** [comment]
Summary of my findings:

* The `seqdiag` plugin works on Xubuntu 21.04 with Zim 0.74.0
  * There's something wrong with your system
    1. Is the output of `which seqdiag` `/usr/local/bin/seqdiag`?
    2. Have you modified your `$PATH`?
    3. Could you run Zim from the repository with `./zim.py --debug`, try to enable/disable the plugin, and then post the output?

* The `zeitgeist` plugin has broken due to changes in Zim ~0.74.0~ _edit:_ it doesn't work with 0.73.5 either!
  * ~Please open an issue on the repository for this~ Oops, sorry, it seems you already have: #787
  * Include the version of Zim with which the plugin still worked; it seems to have broken a while ago
  * `TypeError: on_page_changed() takes 2 positional arguments but 3 were given` is a concrete example of breakage, but there's likely other things wrong, too; hard to tell without logging 

![image](https://user-images.githubusercontent.com/26069787/134476998-a5007faa-8013-4a8e-ac79-c28ca7390b7e.png)

**Comment by Nick222** [comment]
I see file `/usr/bin/seqdiag3`

`which seqdiag` empty output

**Reply by introt** [reply]
Real quick, re: zeitgeist, please update #787 instead of opening a new one

**Reply by introt** [reply]
How did you install it? I installed using `sudo pip install seqdiag` (no 3) and it worked just fine. Could you uninstall it using whatever you installed it with and try again with pip?

**Comment by Nick222** [comment, marked as answer]
I cannot understand - how I install it :(

**Reply by introt** [reply]
I installed it on a new Xubuntu 21.04 system with `sudo pip install seqdiag`. On this system, `which seqdiag` returns `/usr/local/bin/seqdiag`; it should do so on yours, too, but there seems to be something wrong. I do not have a `/usr/bin/seqdiag3` on this system; you've likely installed it using apt. Please do the following:

1. Uninstall the wrong one: `sudo apt purge python3-seqdiag`
2. Install the right one: `sudo pip install seqdiag`
3. What's the output of `which seqdiag` now?

**Reply by introt** [reply]
Alternatively, if you want to use the version installed by apt, you can run `sudo ln -s /usr/bin/seqdiag3 /usr/bin/seqdiag`. This also works, just tested.

**Reply by Nick222** [reply]
> * Uninstall the wrong one: `sudo apt purge python3-seqdiag`
> * Install the right one: `sudo pip install seqdiag`

Now everything OK!
Thank you :)

**Reply by introt** [reply]
You're welcome! I think the plugins should check for `seqdiag3` as well, as they're nowadays the same binary. ~I'll submit a patch.~ _edit: all of these image generator plugins could do with configurable commands, I'll look into this later_ e2:  #1613
